python/cudf/cudf/_lib/spillable_buffer.pyx

```diff
@@ -0,0 +1,448 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+import collections.abc
+import pickle
+import time
+from threading import RLock
+from typing import Any, Tuple, Union
+
+import numpy
+
+import rmm
+
+from cudf.core.buffer import (
+    Buffer,
+    DeviceBufferLike,
+    get_ptr_and_size,
+    is_c_contiguous,
+)
+from cudf.utils.string import format_bytes
+
+from cython.operator cimport dereference
+from libc.stdint cimport uintptr_t
+from libcpp.memory cimport make_shared, shared_ptr, static_pointer_cast
+from libcpp.vector cimport vector
+
+
+cdef shared_ptr[void] create_expose_counter():
+    """Create an expose counter"""
+    return static_pointer_cast[void, int](
+        make_shared[int](42)
+    )
+
+
+cdef class SpillLock:
+    """Disable spilling temporarily for specify buffers"""
+
+    cdef add(self, shared_ptr[void] expose_counter):
+        self._expose_counters.push_back(expose_counter)
+
+
+# TODO: this is not support by PyTorch
```

Comment:
Added:
```
    TODO: this doesn't support libraries such as PyTorch that declare
    the tuple of __cuda_array_interface__["data"] in Cython. In such
    cases, Cython will raise an error because DelayedPointerTuple
    isn't a "real" tuple.
```